src/fdb5/database/Key.cc

```diff
@@ -142,22 +91,21 @@ void Key::decode(eckit::Stream& s) {
     }
 }
 
-void Key::encode(eckit::Stream& s) const {
-    const TypesRegistry* registry = canonical_ ? nullptr : &this->registry();
+void BaseKey::encode(eckit::Stream& s) const {
+    // const TypesRegistry* registry = (registry_ ? registry_.get() : nullptr);
 
     s << keys_.size();
     for (eckit::StringDict::const_iterator i = keys_.begin(); i != keys_.end(); ++i) {
-        s << i->first << (registry ? canonicalise(i->first, i->second) : i->second);
+        s << i->first << canonicalise(i->first, i->second);
     }
 
```

Comment:
readable ?
    for (const auto& [keyword, value] : keys_) { s << keyword << canonicalise(keyword, value); }